biocrnpyler/components_basic.py

```diff
@@ -33,21 +33,13 @@ class DNA(Component):
     """
 
     def __init__(
-            self, name: str, length=0,  # positional arguments
-            mechanisms=None,  # custom mechanisms
-            parameters=None,  # customized parameters
-            attributes=None,
-            initial_conc=None,
-            parameter_warnings = True,
+            self, name: str, length=0, attributes = None,  # positional arguments
             **keywords
     ):
-        self.species = Species(name, material_type="dna",
+        self.species = self.set_species(name, material_type="dna",
```

Comment:
self.set_species(...) is a general method used all the time. the "...)" was there, but on the line below. I have removed the enter, although likely the line will now be too long for pep8.

Reply:
it's not about the line it's about a function is called set species and returns a value, why not setting up self.species inside the function?

Reply:
because that isn't what this function does. It is so Components can set any internal species using a string, component, or Species as an argument. It is used everywhere and has been for a long time.

Reply:
It does is not used for setting component.species.

Reply:
OK, I just try to find oddities during code review, it looked odd, but I see it now (sort of).